controllers/base.go

```diff
@@ -60,16 +75,25 @@ func (c *BaseController) GetSessionCurrentUser() (user *models.User) {
 	return user
 }
 
+func (c *BaseController) RevokeSessionCurrentUser() error {
+	return c.DelSession(CurrentUserKey)
+}
+
 func (c *BaseController) handleAuthorizeRequest() {
-	if c.requireGuestUser && !c.ensureGuestUser() {
+	_, actionName := c.GetControllerAndAction()
+	actionPolicy := c.actionPolicy[actionName]
+
+	if actionPolicy.requireGuestUser && !c.ensureGuestUser() {
 		c.Redirect("/", http.StatusFound)
```

Comment:
@Lahphim Should we redirect to the dashboard? as you said that `dashboard` page is for the authenticated user

Reply:
updated here https://github.com/Lahphim/go-crawler-challenge/pull/51/commits/e4fea30272cd760e4f45313dd12df21693fc07ea